lib/api/1.1/northbound/workflows.js

```diff
@@ -104,34 +104,24 @@ function workflowsRouterFactory (
     }));
 
     /**
-     * @api {get} /api/1.1/workflows/library GET /library
+     * @api {get} /api/1.1/workflows/library/:injectableName GET /library/:injectableName
      * @apiVersion 1.1.0
-     * @apiDescription get list of workflows possible to run
-     * @apiName workflows-library
-     * @apiSuccess {json} workflows list of workflows that are possible to run
-     * @apiGroup workflows
-     */
-
-    router.get('/workflows/library', rest(function () {
-        return workflowApiService.getGraphDefinitions();
-    }));
-
-    /**
-     * @api {get} /api/1.1/workflows/library/:identifier GET /library/:identifier
-     * @apiVersion 1.1.0
-     * @apiDescription get a specific workflow from the library
-     * @apiParam {String} identifier of workflow, must cast to ObjectId
+     * @apiDescription get a specific workflow definition from the library
+     * @apiParam {String} injectableName of workflow definition
      * @apiName workflows-library-item
-     * @apiSuccess {json} workflow the workflow that have the <code>identifier</code>
+     * @apiSuccess {json} workflow the workflow that have the <code>injectableName</code>
      * @apiGroup workflows
      */
 
-    router.get('/workflows/library/:identifier', rest(function (req) {
-        return workflowApiService.getGraphDefinitions(req.params.identifier)
+    router.get('/workflows/library/:injectableName', rest(function (req) {
+        if (req.params.injectableName === '*') {
```

Comment:
In terms of good API design, you're right - we got ourselves into a pickle with this setup, and there's no clean way to escape it with the 1.1 API that doesn't do something to muck up backwards compatibility. The optimal place to resolve this will be the 2.0 API that @brianparry is working on. As a workaround I'm fine with this with updated documentation - maybe an honorable mention in the APIdoc strings above for how to handle what we'd previously intended with a `GET /workflows/library` and relevant notes in the user's guide as well. 

@jlongever if I'm understanding correctly, the two routes is what we intended, but the system wasn't routing the calls to the relevant control code correctly, or routing them to two places at one, which is what we were seeing in #89, but hadn't stumbled over explicitly until we dug out the bugs with the workflow refactor knowledge.